src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/Bjorhus.cpp

```diff
@@ -0,0 +1,488 @@
+      make_not_null(&constraint_char_zero_minus), make_not_null(&char_speeds),
+      face_mesh_velocity, normal_covector, pi, phi, spacetime_metric, coords,
+      gamma1, gamma2, lapse, gauge_source, spacetime_deriv_gauge_source, dt_pi,
+      dt_phi, dt_spacetime_metric, d_pi, d_phi, d_spacetime_metric);
+
+  // Account for moving mesh: char speeds -> cher speeds - n_i v^i_g
+  if (face_mesh_velocity.has_value()) {
+    const auto negative_lambda0 =
+        get(dot_product(normal_covector, *face_mesh_velocity));
+    for (size_t a = 0; a < 4; ++a) {
+      char_speeds.at(a) -= negative_lambda0;
+    }
+  }
+
+  detail::set_dt_v_psi_constraint_preserving(
+      make_not_null(&bc_dt_v_psi), unit_interface_normal_vector,
+      three_index_constraint, char_projected_rhs_dt_v_psi, char_speeds);
+
+  detail::set_dt_v_zero_constraint_preserving(
+      make_not_null(&bc_dt_v_zero), unit_interface_normal_vector,
+      four_index_constraint, char_projected_rhs_dt_v_zero, char_speeds);
+
+  if (type_ == detail::ConstraintPreservingBjorhusType::ConstraintPreserving) {
+    detail::set_dt_v_minus_constraint_preserving(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, incoming_null_one_form,
+        outgoing_null_one_form, incoming_null_vector, outgoing_null_vector,
+        projection_ab, projection_Ab, projection_AB,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, char_speeds);
+  } else if (type_ == detail::ConstraintPreservingBjorhusType::
+                          ConstraintPreservingPhysical) {
+    detail::set_dt_v_minus_constraint_preserving_physical(
+        make_not_null(&bc_dt_v_minus), gamma2, coords, normal_covector,
+        unit_interface_normal_vector, spacetime_unit_normal_vector,
+        incoming_null_one_form, outgoing_null_one_form, incoming_null_vector,
+        outgoing_null_vector, projection_ab, projection_Ab, projection_AB,
+        inverse_spatial_metric, extrinsic_curvature, spacetime_metric,
+        inverse_spacetime_metric, three_index_constraint,
+        char_projected_rhs_dt_v_psi, char_projected_rhs_dt_v_minus,
+        constraint_char_zero_plus, constraint_char_zero_minus, phi, d_phi, d_pi,
+        char_speeds);
+  } else {
+    ERROR("Failed to set dtVMinus. Input option \""
+          << "\" Must be one of ConstraintPreserving or "
+             "ConstraintPreservingPhysical");
+  }
+
+  // Only add corrections at grid points where the char speeds are negative
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_psi), char_speeds.at(0));
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_zero), char_speeds.at(1));
+  helpers::set_bc_corr_zero_when_char_speed_is_positive(
+      make_not_null(&bc_dt_v_minus), char_speeds.at(3));
+
+  // Convert corrections to dt<evolved variables>
+  auto dt_evolved_vars = evolved_fields_from_characteristic_fields(
```

Comment:
Do I understand this correctly that this call is just passing the time derivatives in place of the characteristics and using the result as the time derivatives of the evolved fields?
Is there any danger of other spacetime-dependent factors that have nonvanishing time derivatives that give rise to additional corrections to this calculation?

Reply:
The corrections are applied to the incoming projections of the time derivatives of evolved vars, and *not* to the time derivatives of incoming characteristic variables. Say the evolved fields are `u^A`, and `D^A_B` is the characteristic projection operator. The Bjorhus corrections were derived as:

`D^A_B \partial_t u^B ---> D^A_B \partial_t u^B + corrections^A`

which could also be written as

`\partial_t u^B ----> \partial_t u^B + (D^{-1})^B_A corrections^A`

The `dt_evolved_vars` here is simply the last term `(D^{-1})^B_A corrections^A`. See the discussion around Eq. (62) of [Lindblom et al 2005](https://arxiv.org/abs/gr-qc/0512093).

Reply:
Please add a comment of your answer to the code @prayush

Reply:
This explanation of `dt_evolved_vars` should be made into a comment in the code.